pandas/io/s3.py

```diff
@@ -25,7 +26,15 @@ def get_file_and_filesystem(
     if mode is None:
         mode = "rb"
 
-    fs = s3fs.S3FileSystem(anon=False)
+    # Support customised S3 servers endpoint URL via environment variable
+    # The S3_ENDPOINT should be the complete URL to S3 service following
+    # the format: http(s)://{host}:{port}
+    s3_endpoint = os.environ.get('S3_ENDPOINT')
+    if s3_endpoint:
+        client_kwargs = {'endpoint_url': s3_endpoint}
```

Comment:
Hi xieqihui :) I believe there needs a type annotation, such as
```
        client_kwargs: Optional[Dict[str, str]] = {"endpoint_url": s3_endpoint}
```

Reply:
Thanks for the reminder. I have added the type annotation and format the code.